### PR TITLE
Reg jschema rules to OAC tests

### PR DESCRIPTION
**Pull request by chepil**
SERV-338, SERV-339, SERV-343, SERV-344, SERV-365, SERV-322, SERV-324, SERV-308, SERV-309

### CODE REVIEW

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.minItems_03/minItems.openapi.todo [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+              }
+            }
+          },
+
+          "204": {
+            "description": "",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.minItems_03/minItems.openapi.todo:95]: This array schema is wrong, does not accord to the JSight

```
  204
    [ // { type: "array", nullable: true, minItems: 0 }
    ]
```

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_03/min.max.jst [hunk 2]
```diff
@@ -0,0 +1,29 @@
+JSIGHT 0.3
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_03/min.max.jst:1]: Please, add the same tests with `decimal`.

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_length_03/min.max.length.jst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+JSIGHT 0.3
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_length_03/min.max.length.jst:1]: Please, add tests, where `minLength` and `maxLength` are in the same set of rules.

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.string.jst [hunk 4]
```diff
@@ -0,0 +1,33 @@
+JSIGHT 0.3
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.string.jst:1]: I put the comment here, but it is not related to "nullable.string" tests only. I'd like you to add another test for explicit `nullable: false` rule with all the types. You can add additional file, or add `nullable: false` test to each of the existing "nullable" files.

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.type.object_03/type.object.jst [hunk 5]
```diff
@@ -0,0 +1,11 @@
+JSIGHT 0.3
+
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.type.object_03/type.object.jst:2]: Let's make another test for testing deep cross-structures between arrays and objects. I propose the following case:

```
{
  "object-in-object": {
    "object-in-object": {
      "foo": "bar"
    }
  },
  "short-array-in-object": [
    [ // short-array-in-short-array
      { // object-in-short-array
        "foo": "bar"
      }
    ]
  ],
  "long-array-in-object": [
    "foo",
    { // object-in-long-array
      "foo": "bar"
    },
    [ // short-array-in-long-array
      { // object-in-short-array
        "foo": "bar"
      }
    ],
    [ // long-array-in-long-array
      "foo": "bar",
      { // object-in-long-array
        "foo": "bar",
        "object-in-object": {
          "foo": "bar"
        }
      }
    ]
  ]
}
```

**Review by andrew-platonov — CHANGES_REQUESTED** [review]

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.const_03/const.openapi.todo [hunk 6]
```diff
@@ -0,0 +1,271 @@
+{
+  "openapi": "3.0.3",
+  "info": {
+    "title": "",
+    "version": ""
+  },
+
+  "paths": {
+    "/request/jschema/rules/const": {
```
[the remark this answers is not in the document]
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.const_03/const.openapi.todo:9]: fixed https://github.com/jsightapi/jsight-api-core/pull/15/commits/16b9bfecdfc691dfb17150432fa2a0067f6b8708

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.minItems_03/minItems.openapi.todo (hunk 1 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.minItems_03/minItems.openapi.todo:95]: fixed https://github.com/jsightapi/jsight-api-core/pull/15/commits/2acbd932726b3e5514ef90160b7567b604cbc271

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_03/min.max.jst (hunk 2 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_03/min.max.jst:1]: fixed https://github.com/jsightapi/jsight-api-core/pull/15/commits/dfa26beb5fdb746976e21a45b6a4be0e43f517a8

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_length_03/min.max.length.jst (hunk 3 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.min_max_length_03/min.max.length.jst:1]: fixed https://github.com/jsightapi/jsight-api-core/pull/15/commits/c575a57a5f0981d31c707d6a9dd806659738c94f

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.string.jst (hunk 4 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.string.jst:1]: fixed https://github.com/jsightapi/jsight-api-core/pull/15/commits/427fa5283974abfb514644608f0692f719a51663

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+            "description": "",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "sEnum1": {
+                      "example": "white",
+                      "enum": ["white", "blue", "red"],
+                      "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:59]: nullable is false here

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                    },
+                    "sEnum2": {
+                      "example": -2.1,
+                      "enum": [-3, -2.1, 1.2, true, false, "-3", "0", "1.2", "string", "true"],
+                      "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:64]: nullable is false here

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                    },
+                    "sEnum3": {
+                      "example": null,
+                      "enum": ["white", "blue", "red", null],
+                      "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:69]: nullable is false here

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "anyOf": [
+                      {
+                        "example": "white",
+                        "enum": ["white", "blue", "red"],
+                        "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:90]: nullable is false here

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                      },
+                      {
+                        "example": -2.1,
+                        "enum": [-3, -2.1, 1.2, true, false, "-3", "0", "1.2", "string", "true"],
+                        "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:95]: nullable is false here

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+                      },
+                      {
+                        "example": null,
+                        "enum": ["white", "blue", "red", null],
+                        "nullable": true
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo:100]: nullable is false here

**Review by andrew-platonov — CHANGES_REQUESTED** [review]

**Review by andrew-platonov — APPROVED** [review]

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 7 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 8 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 9 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 10 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 11 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

File: testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo (hunk 12 above)
Comment (chepil) [reply testdata/jsight_0.3/ser-to-openapi/req.jschema.rules.nullable_03/nullable.false.enum.openapi.todo]: https://github.com/jsightapi/jsight-api-core/pull/15/commits/c7c48a17ea72184bb0adb9ebbe483908e1e14df5

**Pull request merged**